src/erlfmt_format.erl

```diff
@@ -307,14 +307,23 @@ binary_op_to_algebra('/', _Meta, {atom, _, _} = Left, {integer, _, _} = Right) -
 binary_op_to_algebra('|', Meta, Left, Right) ->
     %% | does not increase indentation and breaks as a whole
     union_to_algebra(Meta, Left, Right);
-binary_op_to_algebra(Op, Meta0, Left, Right) ->
+binary_op_to_algebra(Op, Meta0, Left0, Right0) ->
     %% don't print parens and comments twice - expr_to_algebra took care of it already
     Meta = erlfmt_scan:delete_annos([parens, pre_comments, post_comments], Meta0),
+    {op, _Meta, Op, Left, Right} = rewrite_assoc(Op, Meta, Left0, Right0),
     binary_op_to_algebra(Op, Meta, Left, Right, ?INDENT).
 
+rewrite_assoc(Op, MetaA, A, {op, MetaBC, Op, B, C} = BC) ->
+    case erlfmt_scan:get_anno(parens, MetaBC, false) of
+        true -> {op, MetaA, Op, A, BC};
+        _ -> rewrite_assoc(Op, MetaA, rewrite_assoc(Op, MetaBC, A, B), C)
```

Comment:
I believe we could remove the complexity in `has_break_between_op` if we tracked the locations here and amended `Meta` nodes. Something like:

```suggestion
        _ -> 
            AB = rewrite_assoc(Op, update_meta_location(MetaBC, A, B), A, B),
            rewrite_assoc(Op, update_meta_location(MetaA, AB, C), AB, C)
```
Given the following helper function
```erlang
update_meta_location(Meta, First, Last) ->
    Meta#{
        location := erlfmt_scan:get_anno(First, location),
        end_location := erlfmt_scan:get_anno(Last, end_location)
    }.
```

Reply:
I will try this, great idea :)

Reply:
This worked out very well, thank you